### DISCUSSION TITLE
How to pass aditional parameters to clickhouse connect?

### DISCUSSION THREAD

**Discussion by leagueofsoups** [apache/superset#24441] [Q&A / Help]
I looking to pass custom send_receive_timeout. Can anybody explain how I can do this? Not working with "Engine Parameters" or DSN args.
https://github.com/ClickHouse/clickhouse-connect/blob/main/clickhouse_connect/driver/httpclient.py#L57
https://clickhouse.com/docs/en/integrations/python#connection-arguments

**Comment by Shriansh2002** [comment]
Try To pass a custom send_receive_timeout in ClickHouse Connect's httpclient.py, you can modify the HTTPClient class constructor to include the send_receive_timeout parameter. Here's an example:

```python
class HTTPClient:
    def __init__(self, url, send_receive_timeout=10):
        self.url = url
        self.send_receive_timeout = send_receive_timeout
        self.session = requests.Session()
        self.session.headers.update({'User-Agent': 'clickhouse-driver'})
```

In this example, the send_receive_timeout parameter is set to a default value of 10 seconds. You can modify this value to suit your needs.
To use this modified HTTPClient class, you can import it into your code and create an instance of it with the desired send_receive_timeout value:

```python
from clickhouse_connect.driver.httpclient import HTTPClient

client = HTTPClient('http://localhost:8123', send_receive_timeout=30)

In this example, the HTTPClient instance is created with a send_receive_timeout value of 30 seconds. This value will be used for all subsequent requests made with this client instance.

```

**Reply by leagueofsoups** [reply]
Hello, but how to do it in superset?
https://github.com/apache/superset/blob/master/superset/db_engine_specs/base.py#L1895-L1902
```
class BasicParametersType(TypedDict, total=False):
    username: str | None
    password: str | None
    host: str
    port: int
    database: str
    query: dict[str, Any]
    encryption: bool
```

**Reply by marcusjfloyd** [reply]
Hey @leagueofsoups, did you ever figure out how to do this. I'm trying to set the timezone, but it's not clear how to do it in superset. I've tried with the engine parameters and DSN as well with no luck.

**Reply by leagueofsoups** [reply]
@marcusjfloyd , hi! I wrote a patch and applied it at the stage of building my container.  It looked bad, but it solved the problem in my production environment. I can't share them for legal reasons.
